### ISSUE TITLE
[BUG] 10.4 and later: $npm_package_config_* variables no longer work for arrays

### ISSUE THREAD

**Issue by Twipped** [npm/cli#7343]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### This issue exists in the latest npm version

- [X] I am using the latest npm

### Current Behavior

In our package.json we have an array of values that we access via environment variables in our npm scripts. Example:
```json
  "scripts": {
    "echo": "echo $npm_package_config_foo $npm_package_config_bar"
  },
  "config": {
    "foo": [
      "a",
      "b",
      "c"
    ],
    "bar": "d e f"
  },
```

Under 10.4 and 10.5, the `$npm_package_config_foo` variable is either empty, or isn't defined.

```sh
➜  /workspace git:(main) ✗ npm run echo 

> javascript@1.0.0 echo
> echo $npm_package_config_foo $npm_package_config_bar

d e f
➜  /workspace git:(main) ✗ npm --version
10.5.0
```

### Expected Behavior

Under npm 10.2 and 10.3, arrays would be concatenated into a space separated list. 

Running `npm run echo` outputs:

```sh
➜  /workspace git:(main) ✗ npm run echo

> javascript@1.0.0 echo
> echo $npm_package_config_foo $npm_package_config_bar

a b c d e f
➜  /workspace git:(main) ✗ npm --version
10.2.4

```

### Steps To Reproduce

I have created a codesandbox with an example. https://codesandbox.io/p/devbox/dnkx8d

Running `nvm i 18.19` will get you npm 10.2.4, and `nvm i 18.20` will get you npm 10.5.0.

In my local debugging I confirmed the change was between 10.3 and 10.4.

### Environment

- npm: 10.5.0
- Node.js: 18.20.1
- OS Name: Linux and macOS
- System Model Name: Macbook Pro M1, and in codesandbox
- npm config:
```ini
; "env" config from environment

cache = "/root/.cache/npm" 
store-dir = "/root/.cache/pnpm" 

; node bin location = /usr/local/share/nvm/versions/node/v18.20.1/bin/node
; node version = v18.20.1
; npm local prefix = /workspace
; npm version = 10.5.0
; cwd = /workspace
; HOME = /root
; Run `npm config ls -l` to show all defaults.
```

**Comment by milaninfy** [comment]
@Twipped It can be accessed as `"echo $npm_package_config_foo_0  $npm_package_config_foo_1  $npm_package_config_foo_2  $npm_package_config_bar"`
This was done here https://github.com/npm/run-script/pull/194

**Comment by Twipped** [comment]
I understand that, but the behavior where the array was coerced into arguments has vanished, and it broke some of our workflows.

**Comment by milaninfy** [comment]
@Twipped This is not a bug and it's expected behaviour and those values can be accessed as `_0`, `_1`, `_2` and so on based on how many values are there in array. This behaviour got changed in v7 and it got fixed in this PR https://github.com/npm/run-script/pull/194

**Comment by Twipped** [comment]
Did you read any of what I wrote? The behavior I'm referencing did not change in v7, it changed in 10.4 with the change of npm version released in node 18.19 and 18.20, and I gave you a reproduction of that exact change.

I am not talking about referencing individual items in the array, I'm talking about referencing the array as a whole. You released a breaking change as a minor update.

**Comment by markrreynolds** [comment]
@milaninfy Prior to the latest patch version change the behavior worked.

I am surprised that no one in the pull request called this interface change out as the breaking change that it is.
